### ISSUE TITLE
Improve nimdoc snippet hyperlinking to documentation

### ISSUE THREAD

**Issue by gradha** [nim-lang/Nim#422]
The nimdoc command can syntax highlight the source code of embedded examples. It would be even better if the embedded code could be processed by the compiler completely to figure out the modules and docstrings and hyperlink documentation like [the_hyperlink_vs_nimrod does](https://github.com/gradha/the_hyperlink_vs_nimrod). See http://forum.nimrod-code.org/t/150 for further idea/information about this.

Basically this would mean preppending `.. code-block:: idetools` to the source code blocks and adding a configuration parameter to make it link to a website or custom base URL.

A nice bonus of this feature is that suddenly all the snippets count as unit testing. No more _Oh, I forgot to update the xxx example_ after changing the API.

**Comment by andreaferretti** [comment]
I think that at least the latter part of this has been implemented in the form of runnable examples.
